charts/portworx/templates/hooks/post-install/px-create-cluster-token.yaml

```diff
@@ -0,0 +1,125 @@
+{{- if (.Values.clusterToken.create) }}
+  {{- $customRegistryURL := .Values.customRegistryURL | default "none" }}
+  {{- $registrySecret := .Values.registrySecret | default "none" }}
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: kube-system
+  name: px-set-cluster-token
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "10"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      {{- if not (eq $registrySecret "none") }}
+      imagePullSecrets:
+        - name: {{ $registrySecret }}
+      {{- end }}
+      restartPolicy: Never
+      serviceAccountName: {{ .Values.clusterToken.serviceAccountName }}
+      containers:
+        - name: post-install-job
+          {{- if eq $customRegistryURL "none" }}
+          image: "bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- else}}
+          image: "{{ $customRegistryURL }}/bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- end }}
+          env:
+            - name: NS
+              value: kube-system
+            - name: KEY
+              value: cluster-wide-secret-key
+          command: ['/bin/bash', '-c']
+          args:
+            - kubectl -n $NS exec -c portworx $(kubectl -n $NS get pods -lname=portworx -o jsonpath='{range .items[*]}{.status.conditions[?(@.type=="Ready")].status}{" "}{.metadata.name}{"\n"}{end}' | grep True | cut -d' ' -f2 | head -n1) -- /opt/pwx/bin/pxctl secrets set-cluster-key --secret $KEY
+      initContainers:
+        - name: post-install-job-init
+          {{- if eq $customRegistryURL "none" }}
+          image: "bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- else}}
+          image: "{{ $customRegistryURL }}/bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- end }}
+          env:
+            - name: NS
+              value: kube-system
+          command: ['/bin/bash', '-c']
+          args:
+            - output=$(kubectl -n $NS get pods -l name=portworx -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}'); while [[ $(echo -n $output | wc -c) -eq 0 ]] || [[ $(echo -n $output | grep -v True) ]]; do sleep 5; output=$(kubectl -n $NS get pods -l name=portworx -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}'); done
+---
+apiVersion: v1
```

Comment:
I am going to leave it as it was before because there is a parameter for service account (`clusterToken.serviceAccountName`). I checked the history and it seems earlier there was a common service account, but later it was split into two. I am not sure about the reason, so going to keep it consistent for now. In future we can deprecate and remove this entire cluster token part.